src/transformers/pipelines/text_classification.py

```diff
@@ -95,30 +125,20 @@ def __call__(self, *args, return_all_scores=None, function_to_apply=None, **kwar
         return_all_scores = return_all_scores if return_all_scores is not None else self.return_all_scores
         function_to_apply = function_to_apply if function_to_apply is not None else self.function_to_apply
 
-        if function_to_apply is None and self.model.config.problem_type is not None:
-            if self.model.config.problem_type == "multi_label_classification":
-                self.function_to_apply = "sigmoid"
-            elif self.model.config.problem_type == "single_label_classification":
-                self.function_to_apply = "softmax"
-
-        def sigmoid(_outputs):
-            return 1.0 / (1.0 + np.exp(-_outputs))
-
-        def softmax(_outputs):
-            maxes = np.max(_outputs, axis=-1, keepdims=True)
-            shifted_exp = np.exp(_outputs - maxes)
-            return shifted_exp / shifted_exp.sum(axis=-1, keepdims=True)
-
-        if function_to_apply == "default":
-            if self.model.config.num_labels == 1:
-                scores = sigmoid(outputs)
-            else:
-                scores = softmax(outputs)
-        elif function_to_apply == "sigmoid":
+        if function_to_apply is None:
+            if self.model.config.problem_type == "multi_label_classification" or self.model.config.num_labels == 1:
+                function_to_apply = ClassificationFunction.SIGMOID
+            elif self.model.config.problem_type == "single_label_classification" or self.model.config.num_labels > 1:
+                function_to_apply = ClassificationFunction.SOFTMAX
+
+        if isinstance(function_to_apply, str):
+            function_to_apply = ClassificationFunction[function_to_apply.upper()]
```

Comment:
Should we try except with clean error message, or is the current exception good enough ?

Reply:
I'm thinking the exception is good enough, but happy to update if you feel strongly

Reply:
No no I think it's fine, I didn't check what the actual message was, in the previous iteration it was explicit so I was wondering if it was ok here. I trust you there.